fastlane/Fastfile

```diff
@@ -166,18 +166,33 @@ def push_pods
   pod_push(path: "PurchasesHybridCommon.podspec", synchronous: true)
 end
 
-def carthage_archive
+def archive
+  framework_filename = "PurchasesHybridCommon.framework"
+  build_app(
+    workspace: "ios/PurchasesHybridCommon/PurchasesHybridCommon.xcworkspace",
+    configuration: "Release",
+    scheme: "PurchasesHybridCommon",
+    silent: true,
+    clean: true,
+    derived_data_path: "derived_data/",
+    output_name: framework_filename,
+    destination: 'generic/platform=iOS'
+  )
+
+  # fastlane gym doesn't fully support .frameworks, so we need to look for the output
+  # in derived data manually. 
   Dir.chdir("..") do
-    # As of Carthage 0.38.0, we still can't archive xcframeworks directly. 
-    # there are also some issues which prevent us from archiving frameworks directly, since
-    # carthage can't deal with architectures for simulators and for Apple Silicon correctly.
-    # We use this script as a workaround until this is fixed. 
-    # https://github.com/Carthage/Carthage/releases/0.38.0
-    sh("./carthage.sh", "build", "--archive", "--platform", "iOS")
-  end
+    output_framework_path = Dir["derived_data/**/#{framework_filename}"].first
+    UI.error("framework not found!") if output_framework_path.nil?
 
-  def is_snapshot_version?(version_name)
-    version_name.end_with?("-SNAPSHOT")
+    # compress using ditto because zip can cause issues with symlinks
+    sh("ditto", "-c", "-k", "--sequesterRsrc", "--keepParent", 
+      output_framework_path, 
+      "#{framework_filename}.zip")
```

Comment:
This was a big issue for purchases-ios with catalyst at one point, so I'm doing it just in case. 
`ditto` seems to be what macOS's Finder uses under the hood when you right-click and select "Compress" and produces identical output